src/wp-includes/rest-api/endpoints/class-wp-rest-pattern-directory-controller.php

```diff
@@ -83,7 +83,14 @@ public function get_items_permissions_check( $request ) {
 	 * @return WP_Error|WP_REST_Response Response object on success, or WP_Error object on failure.
 	 */
 	public function get_items( $request ) {
-		$query_args  = array();
+		// Include an unmodified $wp_version.
+		require ABSPATH . WPINC . '/version.php';
+
+		$query_args = array(
+			'locale'     => get_user_locale(),
+			'user-agent' => 'WordPress/' . $wp_version . '; ' . home_url( '/' ),
+		);
```

Comment:
I think a hash of the final args will need to be included in the transient key, otherwise the wrong cache will be returned when the user changes locales, search terms, etc.

Reply:
The caching happens in a different function, though: `_load_remote_block_patterns()`. Maybe the caching should move to `get_items()` ?

Reply:
That might eventually need to happen anyway, when we enable searching for non-Core patterns?

Reply:
I don't think you want to be setting a user-agent **Query args** here, that nullifies any caching ability of the API. I'm assuming you were intending on setting it as the actual user-agent, which I don't think is needed

I think you'd want to set a `wp_version` field instead, so that it can be included the the `$args` of the URI and be cached there if needed?

Reply:
I also agree, that the caching should probably be in this method, although i haven't looked at where it currently is. Keeping the caching as close to where the args are manipulated seems best to me.

Reply:
Ah, great catch! A `wp-version` arg works for me 👍🏻 

I had just copied what the Events Widget was doing, but forgot that we'll have [nginx caching](https://make.wordpress.org/systems/2021/05/18/patterns-api-maybe-cache-direct-to-proper-upstreams/) on this endpoint instead.

I'm curious why a `user-agent` param in the URL nullifies nginx caching, though. Is it treated different than other query args?

Is it still important to use the "unmodified" version of `$wp_version` in the `wp-version` arg? If so, why? That isn't clear to me from the existing comments in Core.

Reply:
Moved the caching in 67d8e37

I also changed it to be a site transient, since the response should be identical across the network.

Reply:
> I'm curious why a user-agent param in the URL nullifies nginx caching, though. Is it treated different than other query args?

It's because it's a query arg, and not actually setting the user-agent HTTP header like you intended. Sorry I didn't make that clear enough :)

> Is it still important to use the "unmodified" version of $wp_version in the wp-version arg? If so, why? That isn't clear to me from the existing comments in Core.

I believe so, yes. It's mostly an issue for anything that decides not to send content in response to the version header (ie. If it were to report WordPress 1.0, what would the code do? What if it says WordPress 12.3?)

It's included as some security-by-obscurity plugins contaminated the global `$wp_version` variable, that behaviour hasn't suddenly stopped AFAIK, and based on the invalid data that's crept into the stats systems in the last few years, I suspect is still in wide-enough usage.

Reply:
Oh, duh 🤦🏻 , it's because the other query arg contained the `home_url()`, not because it was a query arg vs header 👍🏻